src/javascript/app_2/App/Components/Form/TimePicker/time_picker.jsx

```diff
@@ -0,0 +1,134 @@
+import classNames   from 'classnames';
+import { observer } from 'mobx-react';
+import PropTypes    from 'prop-types';
+import React        from 'react';
+import Dialog       from './dialog.jsx';
+import InputField   from '../input_field.jsx';
+
+class TimePicker extends React.PureComponent {
+    state = {
+        is_open: false,
+        value  : '',
+    };
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleClickOutside);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+
+    toggleDropDown = () => {
+        this.setState({ is_open: !this.state.is_open });
```

Comment:
better to pass function that returns new state when calling setState here: 
`this.setState((state) => ({ is_open : !state.is_open }));`